toolchain.py

```diff
@@ -478,26 +476,7 @@ def report_hook(index, blksize, size):
             try:
                 urlretrieve(url, filename, report_hook)
             except AttributeError:
-                if IS_PY2:
-                    # This is caused by bug in python-future, causing occasional
-                    #     AttributeError: '_fileobject' object has no attribute 'readinto'
-                    # It can be removed if the upstream fix is accepted. See also:
-                    #   * https://github.com/kivy/kivy-ios/issues/322
-                    #   * https://github.com/PythonCharmers/python-future/pull/423
-                    import requests
-
-                    logger.warning("urlretrieve failed. Falling back to request")
-
-                    headers = {'User-agent': 'Mozilla/5.0 (X11; Linux x86_64) '
-                               'AppleWebKit/537.36 (KHTML, like Gecko) '
-                               'Chrome/28.0.1500.71 Safari/537.36'}
-                    r = requests.get(url, headers=headers)
-
-                    with open(filename, "wb") as fw:
-                        fw.write(r.content)
-                    break
-                else:
-                    raise
+                raise
```

Comment:
That actually means we don't even need to catch for `AttributeError` if we just raise it without processing anything

Reply:
Ha! Good point. Will do...